docs/requirements.txt

```diff
@@ -0,0 +1,3 @@
+myst-parser~=1.0
+Sphinx~=6.2
+sphinx-rtd-theme~=1.2
```

Comment:
What are your thoughts on pinning dependencies exactly and relying on dependabot to bump them, dince this is effectively a production deployment.

Reply:
I don't have a lot of experience with dependabot specifically.

I tend not to like pinning exact versions (_i.e._ `==`) unless it is necessary. It feels a bit like you are avoiding keeping your work up to date.

Compatible major versions `~=x.x` feels like a sensible compromise to avoid major version bumps. (That said, I've not personally had too many problems always using latest).

It feels currently low risk because if the site fails to build, CI fails and no deployment happens. That should prompt us to look.

I think you might have more experience with this sort of thing though?

Reply:
That sounds like a good system. I suppose the dependabot idea is that bumping dependency version is always a PR so you can check everything works in CI?

I had not come across `pip-compile` before but that sounds really useful. You get the benefits of pinning _all_ dependencies but only needing to manually specify your direct dependencies. I will need to remember to try that out when I have an opportunity 😄.

Always something we can add later if we feel the need. Ensuring people can rebuild old versions sounds like a good benefit.